addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1270,11 +1270,28 @@ var UrlWidget = InputField.extend({
     /**
      * Urls are links in readonly mode.
      *
+     * You can use attributes to customize its visualization:
+     *
+     * `text` set a static text for the anchor content
+     * `text_field` use another field for the anchor content
+     *
+     * Fallbacks to `value` (the plain URL).
+     *
      * @override
      */
     init: function () {
         this._super.apply(this, arguments);
         this.tagName = this.mode === 'readonly' ? 'a' : 'input';
+        this.anchor_text = this.attrs.text || this.value;
+        // retrieve customized `<a />` text from a field
```

Comment:
Since it is already documented on the docstring, I'm not sure this is useful to repeat it.